src/hooks/coreSDK/useInitSafeCoreSDK.ts

```diff
@@ -1,18 +1,24 @@
 import { useEffect, useState } from 'react'
-import { useCurrentChain } from '@/hooks/useChains'
+import useChainId from '@/hooks/useChainId'
 import useWallet from '../wallets/useWallet'
 import useSafeInfo from '@/hooks/useSafeInfo'
 import { initSafeSDK, setSafeSDK } from '@/hooks/coreSDK/safeCoreSDK'
 
 export const useInitSafeCoreSDK = (): Error | null => {
-  const chain = useCurrentChain()
+  const chainId = useChainId()
   const wallet = useWallet()
   const { safe, safeLoaded } = useSafeInfo()
 
   const [error, setError] = useState<Error | null>(null)
 
   useEffect(() => {
-    if (!safeLoaded || !chain || !wallet || chain.chainId !== wallet.chainId) {
+    if (
+      !safeLoaded ||
+      !wallet?.provider ||
+      !wallet.chainId ||
+      chainId !== wallet.chainId ||
+      safe.chainId !== wallet.chainId
```

Comment:
To summarise our discussion: if you pass the entire `safe` (`SafeInfo`) to `initSafeSDK`, you can get the `chainId`, `address` and `version` without needing to reference the wallet (which may be incorrect). These clauses are then no longer necessary. However, as agreed, I'll leave you to look into this further.

Reply:
After thinking about this, to initialize the Core SDK in every case we need a wallet provider, safe address and safe version. This means it needs to be dependant on the wallet and safe info. I think we can remove the dependency on `chainId` that comes through the URL and instead just make sure that the wallet and safe info have the same chainId.

In that case we can simplify the method signature of `initSafeSDK` and just pass the `SafeInfo` to get all the values we need apart from the wallet provider.

I will add a few tests for `useInitSafeCoreSDK` first and then attempt the refactor.

Reply:
I wouldn't use the entire SafeInfo object tho, it's updated every 15 seconds. It's better to use the individual fields you need.